modules/cfe_testcase/src/time_external_test.c

```diff
@@ -0,0 +1,103 @@
+/*************************************************************************
+**
+**      GSC-18128-1, "Core Flight Executive Version 6.7"
+**
+**      Copyright (c) 2006-2019 United States Government as represented by
+**      the Administrator of the National Aeronautics and Space Administration.
+**      All Rights Reserved.
+**
+**      Licensed under the Apache License, Version 2.0 (the "License");
+**      you may not use this file except in compliance with the License.
+**      You may obtain a copy of the License at
+**
+**        http://www.apache.org/licenses/LICENSE-2.0
+**
+**      Unless required by applicable law or agreed to in writing, software
+**      distributed under the License is distributed on an "AS IS" BASIS,
+**      WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+**      See the License for the specific language governing permissions and
+**      limitations under the License.
+**
+** File: time_external_test.c
+**
+** Purpose:
+**   Functional test of basic External Time Source APIs
+**
+**   Demonstration of how to register and use the UT assert functions.
+**
+*************************************************************************/
+
+/*
+ * Includes
+ */
+
+#include "cfe_test.h"
+
+int32 TestCallbackFunction(void)
+{
+    CFE_FT_Global.Count += 1;
+    return CFE_SUCCESS;
+}
+
+int32 TestCallbackFunction2(void)
+{
+    CFE_FT_Global.Count = 0;
+    return CFE_SUCCESS;
+}
+
+void TestCallback(void)
+{
+    CFE_FT_Global.Count = 1;
+
+    UtPrintf("Testing: CFE_TIME_RegisterSynchCallback, CFE_TIME_UnregisterSynchCallback");
+
+    UtAssert_INT32_EQ(CFE_TIME_RegisterSynchCallback(&TestCallbackFunction), CFE_SUCCESS);
+    UtAssert_INT32_EQ(CFE_TIME_RegisterSynchCallback(&TestCallbackFunction2), CFE_TIME_TOO_MANY_SYNCH_CALLBACKS);
+
+    OS_TaskDelay(2000);
+    if (CFE_FT_Global.Count < 2)
+    {
+        UtAssert_MIR("Timing performance of this system is insufficient to test CFE_TIME_RegisterSynchCallback");
```

Comment:
How about "...likely insufficient..."  Comment might help also, something about the callback only gets called if the elapsed time between tones is within the configured tolerance.  Manual inspection is required to determine if there is an actual failure in registering or executing the callback or if it's a timing issue within the system under test, so the test is not automatically failed here but requires further analysis.